### PR TITLE
Add context parameter to Reconcile function of the Reconciler interface

### PR DESCRIPTION
**Pull request by aorcholski**
## Description

The `Reconcile(ctx)` function of `Reconciler` interface requires context parameter to enforce the use of the context.

ActiveGate Reconcilers used context.TODO() instead of top-level context. Now the context is passed along the call stack:
- authtoken
- capability
- customproperties
- proxy
- statefulset

Additionally the following Reconcilers also pass the context to functions that need it instead of keeping it in the Reconciler struct: 
- connectioninfo
- deploymentmetadata
- dtpullsecret
- webhook_cert_controller

## How can this be tested?
All e2e tests should PASS.

## Checklist

~~- [ ] Unit tests have been updated/added~~
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by waodim — APPROVED** [review]
LGTM

File: src/controllers/dynakube/activegate/internal/proxy/reconciler_test.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func newTestReconcilerWithInstance(client client.Client) *Reconciler {
 func TestReconcileWithoutProxy(t *testing.T) {
 	t.Run(`reconcile dynakube without proxy`, func(t *testing.T) {
 		r := newTestReconcilerWithInstance(fake.NewClientBuilder().Build())
-		err := r.Reconcile()
+		err := r.Reconcile(context.TODO())
```
Comment (0sewa0) [inline src/controllers/dynakube/activegate/internal/proxy/reconciler_test.go:48]: Nitpick: I think we should start using `context.Background` in tests. So we can use `context.TODO` as an actual TODO. (I might have been for using `context.TODO` in tests, and told people to do this, but I was wrong 😬 )

also I should probably add this in the styleguide or somewhere 😅

Comment (0sewa0) [reply src/controllers/dynakube/activegate/internal/proxy/reconciler_test.go:48]: you don't have to update anything you haven't touched! but if we are touching it right now, lets do it properly

Comment (andriisoldatenko) [reply src/controllers/dynakube/activegate/internal/proxy/reconciler_test.go:48]: agree with @0sewa0 ☝️

Comment (aorcholski) [reply src/controllers/dynakube/activegate/internal/proxy/reconciler_test.go]: agree with you :) fixed

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**